clang/lib/CodeGen/CGCall.cpp

```diff
@@ -760,7 +760,7 @@ CodeGenTypes::arrangeLLVMFunctionInfo(CanQualType resultType,
   unsigned CC = ClangCallConvToLLVMCallConv(info.getCC());
   // This is required so SYCL kernels are successfully processed by tools from CUDA. Kernels
   // with a `spir_kernel` calling convention are ignored otherwise.
-  if (CC == llvm::CallingConv::SPIR_KERNEL && CGM.getTriple().isNVPTX() &&
+  if (CC == llvm::CallingConv::SPIR_KERNEL && (CGM.getTriple().isNVPTX() || CGM.getTriple().isNVPTX()) &&
```

Comment:
Is this the intended behavior?
```suggestion
  if (CC == llvm::CallingConv::SPIR_KERNEL && (CGM.getTriple().isNVPTX() || CGM.getTriple().isAMDGCN()) &&
```

Reply:
This is a mistake I made. After testing, the code added here seems unnecessary. I didn’t understand the logic of this piece, and just add a corresponding logic for amdgcn according to nvptx.